test/metrics/metrics_test.go

```diff
@@ -0,0 +1,311 @@
+		require.NoError(t, err)
+
+		// then
+		// check the UserSignup is created
+		userSignup, err = hostAwait.WaitForUserSignup(userSignup.Name)
+		require.NoError(t, err)
+		// confirm that the user is banned
+		assert.Equal(t, toolchainv1alpha1.UserSignupStateLabelValueBanned, userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
+		mur, err := hostAwait.GetMasterUserRecord(wait.WithMurName("testuser" + id))
+		require.NoError(t, err)
+		assert.Nil(t, mur)
+		// verify the metrics
+		metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 1)
+		metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 0) // not provisioned because banned before signup
+		metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 1)
+		metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)  // no user on member1
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+	})
+
+	t.Run("register new user with preexisting ban", func(t *testing.T) {
+		// given
+		metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName, member2Await.ClusterName})
+		hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().Enabled())
+		id := uuid.Must(uuid.NewV4()).String()
+		email := "testuser" + id + "@test.com"
+		banUser(t, hostAwait, email)
+		// get valid generated token for e2e tests. IAT claim is overridden
+		// to avoid token used before issued error.
+		identity0 := authsupport.NewIdentity()
+		emailClaim0 := authsupport.WithEmailClaim(email)
+		token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0)
+		require.NoError(t, err)
+		route := hostAwait.RegistrationServiceURL
+
+		// when calling the signup endpoint with a valid token to initiate a signup process
+		req, err := http.NewRequest("POST", route+"/api/v1/signup", nil)
+		require.NoError(t, err)
+		req.Header.Set("Authorization", "Bearer "+token0)
+		req.Header.Set("content-type", "application/json")
+		resp, err := HTTPClient.Do(req)
+
+		// then
+		require.NoError(t, err)
+		defer Close(t, resp)
+		// verify the metrics
+		metricsAssertion.WaitForMetricDelta(UserSignupsMetric, 0)
+		metricsAssertion.WaitForMetricDelta(UserSignupsApprovedMetric, 0) // not provisioned because banned before signup
+		metricsAssertion.WaitForMetricDelta(UserSignupsBannedMetric, 0)   // banneduser resource was created before signup
+		metricsAssertion.WaitForMetricDelta(MasterUserRecordsPerDomainMetric, 0, "domain", "external")
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)  // no user on member1
+		metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", member2Await.ClusterName) // no user on member2
+	})
+
+}
+
+func TestMetricsWhenUserDisabled(t *testing.T) {
```

Comment:
let's rename this as well to know what is the purpose of the test
```suggestion
func TestWhenUserDisabledThenNoImpactOnMetrics(t *testing.T) {
```